test/indexing/test_setitem.py

```diff
@@ -51,6 +51,15 @@ def test_combined_index_3(self):
 
         np.testing.assert_allclose(x.numpy(), np_data)
 
+    def test_index_has_range(self):
+        np_data = np.ones((3, 4, 5, 6), dtype='int32')
+        x = paddle.to_tensor(np_data)
+
+        np_data[:, range(3), [1, 2, 4]] = 10
+        x[:, range(3), [1, 2, 4]] = 10
+
+        np.testing.assert_allclose(x.numpy(), np_data)
+
```

Comment:
getitem should also be check, eg. `y = x[:, range(3), [1, 2, 4]]`

Reply:
Done, thanks